contract/src/test/java/io/neow3j/contract/Nep5TokenTest.java

```diff
@@ -131,4 +132,32 @@ public void getBalanceOfWallet() throws Exception {
         Nep5Token token = new Nep5Token(GasToken.SCRIPT_HASH, this.neow);
         assertThat(token.getBalanceOf(w), is(new BigInteger("411285799730")));
     }
+
+    @Test(expected = InsufficientFundsException.class)
```

Comment:
hmmm, can you migrate this to the `@Rule` and you've done in the `InvocationTest`?

Reply:
I started to use the `ExpectedException` when I need more granular control over which function call throws the exceptino. In this example it is not necessary because that last function is the only one that will throw the `InsufficientFundsException`.

However, we can definitely change our style to always use `ExpectedException` instead of the annotation on the test method. I see no downside to it. Though, I wouldn't start to adapt it in existing unit tests, that's definitely not priority work.

Reply:
Great! Clear now. 👍